In [21]:
import pandas as mypd
from sklearn.linear_model import LogisticRegression
mydata = mypd.read_csv("/Users/harisuriya/Desktop/mlworkshop/Data/Logistic_Reg.csv")
import statsmodels.api as mysm
from pandas.core import datetools


In [2]:
mydata


Ind_Exp_Act_Score  Tran_Speed_Score  Peer_Comb_Score  Outcome
0                  6.2               9.3              7.4        1
1                  2.6               2.2              8.7        1
2                  9.5               1.5              8.2        1
3                  2.6               5.0              0.4        0
4                 10.0               7.7              7.2        1
5                  0.2               2.3              3.4        0
6                  2.5               2.8              4.6        0
7                  0.4               1.6              5.3        0
8                  0.2               9.8              0.8        0
9                  8.9               6.1              9.2        1
10                 9.3               5.2              2.1        1
11                 9.4               2.1              4.1        1
12                 0.8               5.1              7.5        1
13                 8.2               3.1              0.9        1
14                 6.5               0.7              5.5        1
15                 1.5               5.7              5.9        1
16                 7.3               0.9              4.8        1
17                 3.2               3.3              3.0        0
18                 6.9               9.4              9.3        1
19                 6.8               9.8              8.3        1
20                 3.6               4.4              7.1        1
21                 0.6               5.9              0.2        0
22                 2.4               3.2              8.2        1
23                 7.2               1.6              5.8        1
24                 3.7               6.6              0.7        0
25                 9.4               9.7              6.0        1
26                 6.5               8.2              8.3        1
27                 6.1               2.0              4.5        1
28                 7.1               1.9              4.8        1
29                 1.2               6.7              7.6        1
..                 ...               ...              ...      ...
950                5.1               4.8              5.7        1
951                1.2               1.1              4.2        0
952                1.5               0.1              7.1        0
953                4.3               7.0              9.0        1
954                5.8               0.1              9.8        1
955                3.1               8.1              3.1        1
956                1.5               5.7              0.4        0
957                9.6               5.9              9.8        1
958                8.2               8.3              7.9        1
959                1.9               4.4              4.2        0
960                6.6               7.5              0.2        1
961                3.7               8.9              0.6        1
962                2.8               6.3              1.3        0
963                7.6               2.0              8.8        1
964                6.8               9.6              1.1        1
965                2.4               2.8              9.0        1
966                9.6               8.7              8.8        1
967                9.2               2.4              7.6        1
968                7.5               2.3              7.9        1
969                1.7               9.8              8.0        1
970                6.8               6.9              2.1        1
971                1.4               6.0              9.4        1
972                1.3               0.8              5.1        0
973                6.7              10.0              0.2        1
974                7.3               2.9              6.0        1
975                6.7               2.7              1.8        0
976                8.3               9.7              5.6        1
977                2.3               0.7            

In [3]:
x = mydata[["Ind_Exp_Act_Score", "Tran_Speed_Score", "Peer_Comb_Score"]]


In [4]:
x

Ind_Exp_Act_Score  Tran_Speed_Score  Peer_Comb_Score
0                  6.2               9.3              7.4
1                  2.6               2.2              8.7
2                  9.5               1.5              8.2
3                  2.6               5.0              0.4
4                 10.0               7.7              7.2
5                  0.2               2.3              3.4
6                  2.5               2.8              4.6
7                  0.4               1.6              5.3
8                  0.2               9.8              0.8
9                  8.9               6.1              9.2
10                 9.3               5.2              2.1
11                 9.4               2.1              4.1
12                 0.8               5.1              7.5
13                 8.2               3.1              0.9
14                 6.5               0.7              5.5
15                 1.5               5.7              5.9
16                 7.3               0.9              4.8
17                 3.2               3.3              3.0
18                 6.9               9.4              9.3
19                 6.8               9.8              8.3
20                 3.6               4.4              7.1
21                 0.6               5.9              0.2
22                 2.4               3.2              8.2
23                 7.2               1.6              5.8
24                 3.7               6.6              0.7
25                 9.4               9.7              6.0
26                 6.5               8.2              8.3
27                 6.1               2.0              4.5
28                 7.1               1.9              4.8
29                 1.2               6.7              7.6
..                 ...               ...              ...
950                5.1               4.8              5.7
951                1.2               1.1              4.2
952                1.5               0.1              7.1
953                4.3               7.0              9.0
954                5.8               0.1              9.8
955                3.1               8.1              3.1
956                1.5               5.7              0.4
957                9.6               5.9              9.8
958                8.2               8.3              7.9
959                1.9               4.4              4.2
960                6.6               7.5              0.2
961                3.7               8.9              0.6
962                2.8               6.3              1.3
963                7.6               2.0              8.8
964                6.8               9.6              1.1
965                2.4               2.8              9.0
966                9.6               8.7              8.8
967                9.2               2.4              7.6
968                7.5               2.3              7.9
969                1.7               9.8              8.0
970                6.8               6.9              2.1
971                1.4               6.0              9.4
972                1.3               0.8              5.1
973                6.7              10.0              0.2
974                7.3               2.9              6.0
975                6.7               2.7              1.8
976                8.3               9.7              5.6
977                2.3               0.7              5.5
978                0.9               4.8              3.0
979                2.5               0.0              7.4

[980 rows x 3 columns]

In [5]:
y=mydata.Outcome

In [6]:
y

0      1
1      1
2      1
3      0
4      1
5      0
6      0
7      0
8      0
9      1
10     1
11     1
12     1
13     1
14     1
15     1
16     1
17     0
18     1
19     1
20     1
21     0
22     1
23     1
24     0
25     1
26     1
27     1
28     1
29     1
      ..
950    1
951    0
952    0
953    1
954    1
955    1
956    0
957    1
958    1
959    0
960    1
961    1
962    0
963    1
964    1
965    1
966    1
967    1
968    1
969    1
970    1
971    1
972    0
973    1
974    1
975    0
976    1
977    0
978    0
979    0
Name: Outcome, Length: 980, dtype: int64

In [7]:
x["Intercept"]=1

In [8]:
x

Ind_Exp_Act_Score  Tran_Speed_Score  Peer_Comb_Score  Intercept
0                  6.2               9.3              7.4          1
1                  2.6               2.2              8.7          1
2                  9.5               1.5              8.2          1
3                  2.6               5.0              0.4          1
4                 10.0               7.7              7.2          1
5                  0.2               2.3              3.4          1
6                  2.5               2.8              4.6          1
7                  0.4               1.6              5.3          1
8                  0.2               9.8              0.8          1
9                  8.9               6.1              9.2          1
10                 9.3               5.2              2.1          1
11                 9.4               2.1              4.1          1
12                 0.8               5.1              7.5          1
13                 8.2               3.1              0.9          1
14                 6.5               0.7              5.5          1
15                 1.5               5.7              5.9          1
16                 7.3               0.9              4.8          1
17                 3.2               3.3              3.0          1
18                 6.9               9.4              9.3          1
19                 6.8               9.8              8.3          1
20                 3.6               4.4              7.1          1
21                 0.6               5.9              0.2          1
22                 2.4               3.2              8.2          1
23                 7.2               1.6              5.8          1
24                 3.7               6.6              0.7          1
25                 9.4               9.7              6.0          1
26                 6.5               8.2              8.3          1
27                 6.1               2.0              4.5          1
28                 7.1               1.9              4.8          1
29                 1.2               6.7              7.6          1
..                 ...               ...              ...        ...
950                5.1               4.8              5.7          1
951                1.2               1.1              4.2          1
952                1.5               0.1              7.1          1
953                4.3               7.0              9.0          1
954                5.8               0.1              9.8          1
955                3.1               8.1              3.1          1
956                1.5               5.7              0.4          1
957                9.6               5.9              9.8          1
958                8.2               8.3              7.9          1
959                1.9               4.4              4.2          1
960                6.6               7.5              0.2          1
961                3.7               8.9              0.6          1
962                2.8               6.3              1.3          1
963                7.6               2.0              8.8          1
964                6.8               9.6              1.1          1
965                2.4               2.8              9.0          1
966                9.6               8.7              8.8          1
967                9.2               2.4              7.6          1
968                7.5               2.3              7.9          1
969                1.7               9.8              8.0          1
970                6.8               6.9              2.1          1
971                1.4               6.0              9.4          1
972                1.3               0.8              5.1          1
973                6.7              10.0              0.2          1
974                7.3               2.9              6.0          1
975                6.7               2.7              1.8          1
976

In [9]:
mymodel = mysm.Logit(y,x)

In [10]:
myresult = mymodel.fit()

Optimization terminated successfully.
         Current function value: 0.064710
         Iterations 12


In [12]:
myresult.summary2()

<class 'statsmodels.iolib.summary2.Summary'>
"""
                           Results: Logit
====================================================================
Model:                 Logit             No. Iterations:    12.0000 
Dependent Variable:    Outcome           Pseudo R-squared:  0.890   
Date:                  2018-06-25 23:16  AIC:               134.8326
No. Observations:      980               BIC:               154.3828
Df Model:              3                 Log-Likelihood:    -63.416 
Df Residuals:          976               LL-Null:           -577.85 
Converged:             1.0000            Scale:             1.0000  
--------------------------------------------------------------------
                   Coef.   Std.Err.    z    P>|z|   [0.025   0.975] 
--------------------------------------------------------------------
Ind_Exp_Act_Score   2.7957   0.3554  7.8674 0.0000   2.0992   3.4922
Tran_Speed_Score    2.7532   0.3428  8.0325 0.0000   2.0814   3.4249
Peer_Comb_Score     3.5153   0.4343  8.0949 0.0000   2.6642   4.3664
Intercept         -35.5062   4.4062 -8.0583 0.0000 -44.1421 -26.8702
====================================================================

"""

to check the model based on following things
   LLR(log likelyhood ratio test)  p value <0.05 
     value is 9.799e-223 this is significant
Accuary
pseudo R-squ: >0.6
    here the value is 0.8903
all the of x ,p[z] is zero all the x values are required 
p(y) = e pow x (2.7957+ 2.7532x1+3.513x2-35.5062x3)/1+(2.7957+ 2.7532x1+3.513x2-35.5062x3)

In [13]:
pred=myresult.predict(x)

In [14]:
pred

0      1.000000e+00
1      9.997758e-01
2      1.000000e+00
3      2.117318e-06
4      1.000000e+00
5      5.800809e-08
6      9.588507e-03
7      1.174995e-05
8      5.744115e-03
9      1.000000e+00
10     9.999949e-01
11     9.999828e-01
12     9.992050e-01
13     2.926327e-01
14     9.806875e-01
15     9.940641e-01
16     9.859917e-01
17     9.785382e-04
18     1.000000e+00
19     1.000000e+00
20     9.999911e-01
21     4.658289e-08
22     9.998551e-01
23     9.999187e-01
24     1.066135e-02
25     1.000000e+00
26     1.000000e+00
27     9.464956e-01
28     9.984188e-01
29     9.999978e-01
           ...     
950    9.999939e-01
951    5.810119e-07
952    2.286325e-03
953    1.000000e+00
954    9.999998e-01
955    9.982713e-01
956    6.717042e-07
957    1.000000e+00
958    1.000000e+00
959    3.502416e-02
960    9.865784e-01
961    8.100476e-01
962    3.130821e-03
963    1.000000e+00
964    9.999990e-01
965    9.999738e-01
966    1.000000e+00
967    1.000000e+00
968    1.000000e+00


In [15]:
myoutput = mypd.DataFrame(pred)

In [16]:
myoutput

0
0    1.000000e+00
1    9.997758e-01
2    1.000000e+00
3    2.117318e-06
4    1.000000e+00
5    5.800809e-08
6    9.588507e-03
7    1.174995e-05
8    5.744115e-03
9    1.000000e+00
10   9.999949e-01
11   9.999828e-01
12   9.992050e-01
13   2.926327e-01
14   9.806875e-01
15   9.940641e-01
16   9.859917e-01
17   9.785382e-04
18   1.000000e+00
19   1.000000e+00
20   9.999911e-01
21   4.658289e-08
22   9.998551e-01
23   9.999187e-01
24   1.066135e-02
25   1.000000e+00
26   1.000000e+00
27   9.464956e-01
28   9.984188e-01
29   9.999978e-01
..            ...
950  9.999939e-01
951  5.810119e-07
952  2.286325e-03
953  1.000000e+00
954  9.999998e-01
955  9.982713e-01
956  6.717042e-07
957  1.000000e+00
958  1.000000e+00
959  3.502416e-02
960  9.865784e-01
961  8.100476e-01
962  3.130821e-03
963  1.000000e+00
964  9.999990e-01
965  9.999738e-01
966  1.000000e+00
967  1.000000e+00
968  1.000000e+00
969  1.000000e+00
970  9.999490e-01
971  1.000000e+00
972  7.959907e-06
973  9.999895e-01
974  9.999992e-01
975  4.681072e-02
976  1.000000e+00
977  4.036541e-04
978  9.815091e-05
979  7.561211e-02

[980 rows x 1 columns]

In [17]:
myoutput.to_csv("E:/ML_Workshop/Data/output.csv")


FileNotFoundError: [Errno 2] No such file or directory: 'E:/ML_Workshop/Data/output.csv'

In [18]:
myoutput

0
0    1.000000e+00
1    9.997758e-01
2    1.000000e+00
3    2.117318e-06
4    1.000000e+00
5    5.800809e-08
6    9.588507e-03
7    1.174995e-05
8    5.744115e-03
9    1.000000e+00
10   9.999949e-01
11   9.999828e-01
12   9.992050e-01
13   2.926327e-01
14   9.806875e-01
15   9.940641e-01
16   9.859917e-01
17   9.785382e-04
18   1.000000e+00
19   1.000000e+00
20   9.999911e-01
21   4.658289e-08
22   9.998551e-01
23   9.999187e-01
24   1.066135e-02
25   1.000000e+00
26   1.000000e+00
27   9.464956e-01
28   9.984188e-01
29   9.999978e-01
..            ...
950  9.999939e-01
951  5.810119e-07
952  2.286325e-03
953  1.000000e+00
954  9.999998e-01
955  9.982713e-01
956  6.717042e-07
957  1.000000e+00
958  1.000000e+00
959  3.502416e-02
960  9.865784e-01
961  8.100476e-01
962  3.130821e-03
963  1.000000e+00
964  9.999990e-01
965  9.999738e-01
966  1.000000e+00
967  1.000000e+00
968  1.000000e+00
969  1.000000e+00
970  9.999490e-01
971  1.000000e+00
972  7.959907e-06
973  9.999895e-01
974  9.999992e-01
975  4.681072e-02
976  1.000000e+00
977  4.036541e-04
978  9.815091e-05
979  7.561211e-02

[980 rows x 1 columns]